### ISSUE TITLE
No power sensor plot even when configured

### ISSUE THREAD

**Issue by jimbolimbo3** [lgarciamarrero92/ha-nilm#4]
I configured correctly the power sensor plot:

<img width="1865" height="894" alt="Image" src="https://github.com/user-attachments/assets/863a6224-7925-45d0-b667-98f2c682e266" />

and proceeded with steps:

<img width="1840" height="771" alt="Image" src="https://github.com/user-attachments/assets/e423a0ae-9338-4c75-85f6-5f488ed2f1c1" />

<img width="1855" height="891" alt="Image" src="https://github.com/user-attachments/assets/1182f630-58af-4c96-a53c-4ed0f47daa09" />

But I get no plot when looking for power for the home appliance:

<img width="1863" height="945" alt="Image" src="https://github.com/user-attachments/assets/64f60f5d-536e-4268-a671-b736e9e59fe0" />

Do you need logs?

Thanks

**Comment by lgarciamarrero92** [comment]
Thanks for the report. To help narrow down the cause, could you please give a few details:

1. What unit does the selected mains sensor use in Home Assistant: `W` or `kW`?
2. What is the approximate update frequency of that sensor: `1s`, `5s`, `10s`, `30s`, `1 min`, etc.?
3. In the next training step, when you choose the appliance ground-truth sensor, does the same problem happen there too?

**Comment by johngrange** [comment]
Hi,

I get exactly the same problem.  My sensor uses kW, entity id is sensor.power_sensor.total_power.  It appears to be reporting approx every 15 seconds.

I get the same blank graph on all the training screens, but the dashboard screen looks OK, as per the diagrams above, so yes, same problem when choosing the appliance ground truth sensor.

There appears to be sensor data sent to front end in the transition from the dashboard to training page - i get the following in the browser development tools console:
DEBUG: Raw data from /history: 
Array [ (9808) […] ]
​0: Array(9808) [ {…}, {…}, {…}, … ]
​[0…99]
​0: Object { x: "2026-05-26T17:23:39.000Z", y: 2.045 }
​1: Object { x: "2026-05-26T17:23:39.820Z", y: 2.045 }
​2: Object { x: "2026-05-26T17:23:39.821Z", y: 0.447 }
​3: Object { x: "2026-05-26T17:23:55.010Z", y: 0.447 }
​4: Object { x: "2026-05-26T17:23:55.011Z", y: 2.078 }

so there are readings and they have y values...

**Comment by lgarciamarrero92** [comment]
This should now be fixed in the latest release.

The issue was caused by power sensors reported in `kW` not being normalized to `W` in the training and live disaggregation flows. Support for `kW` sensors has now been added, so both training and real-time inference should handle those sensors correctly.
